### PR TITLE
Term finance  integration

### PR DESCRIPTION
**Pull request by aazhou1**
Integrating Term Finance auction offer locking and Term Finance TermRepoToken Redemption into Boring Vault

### CODE REVIEW

File: src/base/DecodersAndSanitizers/TermFinanceDecoderAndSanitizer.sol [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+contract TermFinanceDecoderAndSanitizer is BaseDecoderAndSanitizer {
+    //============================== TERM FINANCE ===============================
+
+    constructor(address _boringVault) BaseDecoderAndSanitizer(_boringVault) {}
+
+    function lockOffers(DecoderCustomTypes.TermAuctionOfferSubmission[] calldata offerSubmissions) pure virtual external returns (bytes memory addressesFound) {
+        for (uint256 i = 0; i < offerSubmissions.length; i++) {
+            addressesFound = abi.encodePacked(
+                addressesFound,
+                offerSubmissions[i].offeror //TODO: Sanitize offeror as vault address?
```
Comment (crispymangoes) [inline src/base/DecodersAndSanitizers/TermFinanceDecoderAndSanitizer.sol:15]: I would actually prefer to just report out the offeror address like you are doing. I want to try and move away from Decoders being tied to specific Boring Vaults as its more flexible is any boring vault can use them.

**Review by crispymangoes — COMMENTED** [review]
Yes you are on the right path, this looks great!

File: src/base/DecodersAndSanitizers/TermFinanceDecoderAndSanitizer.sol (hunk 1 above)
Comment (aazhou1) [reply src/base/DecodersAndSanitizers/TermFinanceDecoderAndSanitizer.sol:15]: what is "report out"?

Comment (aazhou1) [reply src/base/DecodersAndSanitizers/TermFinanceDecoderAndSanitizer.sol]: TODO removed

**Comment by crispymangoes** [comment]
I have pulled the code into a new PR and addressed merge conflicts. So this PR code will be integrated [here](https://github.com/Se7en-Seas/boring-vault/pull/146)

**Pull request closed without merging**